client/src/main/java/io/pravega/client/stream/impl/EventStreamWriterImpl.java

```diff
@@ -425,7 +424,7 @@ public void noteTime(long timestamp) {
                                              .stream()
                                              .collect(Collectors.toMap(e -> e.getKey(),
                                                                        e -> e.getValue().getLastObservedWriteOffset()));
-        Position position = new PositionImpl(offsets);
+        WriterPosition position = new WriterPosition(offsets);
```

Comment:
What is the difference between Position and WriterPosition. 
I dont see range included as part of writerPosition.

Reply:
Ok, you have included Range information in PositionImpl. 
But that is a super set of WriterPosition. So what is the need for writer's position object? We could have range information available with the writers too via PositionImpl. 
Do we see it diverging from Position object in future. 
Yes, its true that a writer may include multiple segments that overlap with each other. And in such cases it will only have its own assignment of subset of ranges to segments. So should we include that as part of writer position?

Reply:
We could include this information. As you note the information is not maintained because of the overlap possibility. So we would have to create a separate map/list on the writer to track the original ranges for segments, then we could take the segmentId and assemble it to a segment object from this mapping it stored from the controller, to create a position object which only lives for a single stack frame as it is immediately passed to the controller API which doesn't want Range information. So we'd then have to write another layer of code to remove it. (Or change the wire protocol to send it anyway).

So that would introduce a fair bit of code and runtime overhead that doesn't really serve any purpose.